sentry-core/src/main/java/io/sentry/core/EventCachedEventFireAndForgetIntegration.java

```diff
@@ -0,0 +1,125 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.Charset;
+import java.util.concurrent.Executors;
+
+public final class EventCachedEventFireAndForgetIntegration implements Integration {
+  private static final Charset UTF_8 = Charset.forName("UTF-8");
+
+  @Override
+  public void register(IHub hub, SentryOptions options) {
+    String cachedDir = options.getCacheDirPath();
+    if (cachedDir == null) {
+      logIfNotNull(
+          options.getLogger(), SentryLevel.WARNING, "No cache dir path is defined in options.");
+      return;
+    }
+
+    File outbox = new File(options.getCacheDirPath());
+    try {
+      Executors.callable(
+              () -> {
+                SendCachedFiles(hub, options.getLogger(), options.getSerializer(), outbox);
+              })
+          .call();
+    } catch (Exception e) {
+      logIfNotNull(
+          options.getLogger(),
+          SentryLevel.ERROR,
+          "Failed trying to send cached events at %s",
+          outbox);
+    }
+  }
+
+  private static void SendCachedFiles(
+      IHub hub, ILogger logger, ISerializer serializer, File directory) {
+    if (!directory.exists()) {
+      logIfNotNull(
+          logger,
+          SentryLevel.INFO,
+          "Directory '%s' doesn't exist. Nothing no cached events to send.",
+          directory.getAbsolutePath());
+      return;
+    }
+    if (!directory.isDirectory()) {
+      logIfNotNull(
+          logger,
+          SentryLevel.ERROR,
+          "Cache dir %s is not a directory.",
+          directory.getAbsolutePath());
+      return;
+    }
+
+    logIfNotNull(
+        logger,
+        SentryLevel.DEBUG,
+        "Processing %d items from cache dir %s",
+        directory.length(),
+        directory.getAbsolutePath());
+
+    for (File file : directory.listFiles()) {
+      // TODO: postfix should ve a const, shared with the caching code (still to be merged)
+      if (!file.getName().endsWith(".sentry-event")) {
+        logIfNotNull(
+            logger,
+            SentryLevel.DEBUG,
+            "File '%s' doesn't match extension expected.",
+            file.getName());
+        continue;
+      }
+
+      if (!directory.isFile()) {
+        logIfNotNull(logger, SentryLevel.DEBUG, "'%s' is not a file.", directory.getAbsolutePath());
+        return;
+      }
+
+      if (!file.canRead()) {
+        logIfNotNull(logger, SentryLevel.WARNING, "File '%s' cannot be read.", file.getName());
+        safeDelete(file, "which can't be read but has expected file extension.", logger);
+        continue;
+      }
+
+      CachedEvent hint = new CachedEvent();
+      try (Reader reader =
+          new BufferedReader(new InputStreamReader(new FileInputStream(file), UTF_8))) {
+        SentryEvent event = serializer.deserializeEvent(reader);
+        // TODO: Run a sync capture? The SentryId returned could be the indication to whether delete
```

Comment:
I'd say only to enqueue it, as `EventSender` has `IEventCache` and it's business logic to `eventCache.discard(event)` if so, I'd say we fix the business logic in there if we need to.

Reply:
I need to bypass that logic. Before it discards, it stores and we dont' want this file to be stored again.